### DISCUSSION TITLE
No connection to MQTT

### DISCUSSION THREAD

**Discussion by jklimpel** [softwarecrash/EPEver2MQTT#45] [General]
Hi,
I have problems connecting EPEver2MQTT to the MQTT server. 
I can see data from EPEver on the WEB frontend (see picture), but the connection to the MQTT doesn't work.
Access data for the MQTT is correct and has been entered several times. MQTT server also runs correctly and I have already used it several times for other applications.
Does anyone have an idea?

![EPEver-1](https://github.com/softwarecrash/EPEver2MQTT/assets/85573793/765e74b1-2855-4519-9bbf-f651eed1a236)
![EPEver-2](https://github.com/softwarecrash/EPEver2MQTT/assets/85573793/0a1c938d-b3a2-4404-b2ab-27607cd2da02)

Jurgen

**Comment by softwarecrash** [comment]
> Hi, I have problems connecting EPEver2MQTT to the MQTT server. I can see data from EPEver on the WEB frontend (see picture), but the connection to the MQTT doesn't work. Access data for the MQTT is correct and has been entered several times. MQTT server also runs correctly and I have already used it several times for other applications. Does anyone have an idea?
> 
> ![EPEver-1](https://private-user-images.githubusercontent.com/85573793/333875436-765e74b1-2855-4519-9bbf-f651eed1a236.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MjQxMDcsIm5iZiI6MTcxNjcyMzgwNywicGF0aCI6Ii84NTU3Mzc5My8zMzM4NzU0MzYtNzY1ZTc0YjEtMjg1NS00NTE5LTliYmYtZjY1MWVlZDFhMjM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDExNDMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4MDhiNDJhMTBhNjZiYzk2M2EyNTQ2OGI5ZjUzZjUyNDdkYWZmZWFiNTJmOWM3OGIxYjIwN2UxNzEyMDBmM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kNP91dkzmoVaEu6PnN9Mt7SGgI00_rBxkxrvKGhdHXM) ![EPEver-2](https://private-user-images.githubusercontent.com/85573793/333875447-0a1c938d-b3a2-4404-b2ab-27607cd2da02.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MjQxMDcsIm5iZiI6MTcxNjcyMzgwNywicGF0aCI6Ii84NTU3Mzc5My8zMzM4NzU0NDctMGExYzkzOGQtYjNhMi00NDA0LWIyYWItMjc2MDdjZDJkYTAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDExNDMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkYjM3YWJkYWVkNTY4NmY3OTY1N2M3ODhiYjAyOTQxMWQ0YWVhMDNiNTBhZmFhYWUwOTZhZGUxOGU5ZGRlZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1gMCR1c8MlXtDwwfALOMJgNYmeG0vSWlZ5M2T9BQO-M)
> 
> Jurgen

you  use HomeAssistant?
can you provide a log from your mqtt server, and take a look at the web serial what it tells about the mqtt connection?

**Reply by jklimpel** [reply]
Yes I'm using HomeAssistant
The web serial show nothing about MQTT
![EPEver-3](https://github.com/softwarecrash/EPEver2MQTT/assets/85573793/c747e169-2e60-40c5-9c2c-61d311e0e59e)

But the LED on the ESP8266 flashes 3 times, that means no connection to the MQTT server.
There is nothing in the MQTT log that indicates the EPEver.

**Reply by softwarecrash** [reply]
> Yes I'm using HomeAssistant The web serial show nothing about MQTT ![EPEver-3](https://private-user-images.githubusercontent.com/85573793/333876674-c747e169-2e60-40c5-9c2c-61d311e0e59e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3MjQ5NTEsIm5iZiI6MTcxNjcyNDY1MSwicGF0aCI6Ii84NTU3Mzc5My8zMzM4NzY2NzQtYzc0N2UxNjktMmU2MC00MGM1LTljMmMtNjFkMzExZTBlNTllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDExNTczMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyOTI2MWM1NzlhYmFjMzBmZjA1NDA0N2E2NTA3YWJiNjY0MTIzYzRkYTlmMmE0YzQ2NTRlNmUxZTZjNjNlMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cL3scY53OI3NUFTN9phB7vcRBIDw_GJGjcSt4oxJgHE)
> 
> But the LED on the ESP8266 flashes 3 times, that means no connection to the MQTT server. There is nothing in the MQTT log that indicates the EPEver.

it looks like your device 2,3 are not response. so when not all devices response the runner noticed a fail and doesnt send any data. when you have only one device, set it in the esp settings to 1

**Reply by jklimpel** [reply]
OK the web serial now looks nice
![EPEver-4](https://github.com/softwarecrash/EPEver2MQTT/assets/85573793/edd2b3da-e560-4edc-a4f1-68c0fb59d288)

But no connection to MQTT, LED flashes 3 times.

**Reply by softwarecrash** [reply]
then check your log from your mqtt broker if there some login from the esp or sometings

**Reply by jklimpel** [reply]
sorry for the stupid question, the username was wrong. I re-entered the password 100 times and overlooked the username.
Everything works, thank you very much.
I'm still a little ashamed now.

Jürgen

**Reply by softwarecrash** [reply]
whatever, it works :)
